src/Converter/Extensions/InstagramConverter.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Predmond\HtmlToAmp\Converter\Extensions;
+
+use League\Event\EventInterface;
+use Predmond\HtmlToAmp\ElementInterface;
+use Predmond\HtmlToAmp\Converter\ConverterInterface;
+
+class InstagramConverter implements ConverterInterface
+{
+    public function hasInstagram(ElementInterface $element)
+    {
+        $classes = explode(' ', $element->getAttribute('class'));
+        return in_array('instagram-media', $classes);
+    }
+
+    public function findTheCode($element) 
+    {
+        $href = $element->getAttribute('href');
+        if ($href != '') {
+            $parts = explode('/', $href);
+            if (in_array('www.instagram.com', $parts)) {
+                return $parts[count($parts)-2];
+            }
+        } 
+
+        foreach($element->getChildren() as $child) {
+            $shortcode = $this->findTheCode($child);
+            if ($shortcode !== null) {
+                return $shortcode;
+            }
+        }
+
+        return null; 
+                
+    }
+
+    public function getAmpInstagram(ElementInterface $element, $shortcode) {
+        $attrs = [
+            'layout' => "responsive",
+            'width' => 600,
+            'height' => 384,
+            'data-shortcode' => $shortcode
+        ];
+
+        return $element->createWritableElement('amp-instagram', $attrs);
+    }
+
+    public function handleInstagram( EventInterface $event, ElementInterface $element) {
+
+        if ($this->hasInstagram($element) == false) {
```

Comment:
Instead of `hasInstagram` I prefer `shouldBeConverted($element)` because it is more clear the intention. You can combine these two `if` statements and nest the `hasChildren()` check in the `shouldBeConverted()` method:

```php
if ($this->shouldBeConverted() === false) {
    return;
}
```